### PR TITLE
Correct meaning of 'start' and 'end' text align for LTR text

### LINKED ISSUE #11586: Text alignment wrong for start and end
**Describe the bug**
`start` and `end` text-align are swapped.
https://github.com/openlayers/openlayers/blob/705e328fa096ee3f63fa3c609e55a32315c50c0e/src/ol/render/canvas/TextBuilder.js#L31-L36
**To Reproduce**
1. Copy+paste the data url to get a test page (CTRL + U to view source):
`data:text/html;base64,PCFET0NUWVBFIGh0bWw+CjxodG1sPgo8aGVhZD4KCTx0aXRsZT5UZXh0IGFsaWdubWVudDwvdGl0bGU+Cgk8bWV0YSBjaGFyc2V0PSJVVEYtOCIgLz4KPC9oZWFkPgo8Ym9keT4KCTxkaXYgc3R5bGU9InRleHQtYWxpZ246IGxlZnQiPmxlZnQ8L2Rpdj4KCTxkaXYgc3R5bGU9InRleHQtYWxpZ246IHN0YXJ0Ij5zdGFydDwvZGl2PgoJPGRpdiBzdHlsZT0idGV4dC1hbGlnbjogcmlnaHQiPnJpZ2h0PC9kaXY+Cgk8ZGl2IHN0eWxlPSJ0ZXh0LWFsaWduOiBlbmQiPmVuZDwvZGl2Pgo8L2JvZHk+CjwvaHRtbD4=`
2. Compare the start / left and end / right values in the page with the code in the `TEXT_ALIGN` enum and / or this example:
https://openlayers.org/en/main/examples/vector-labels.html  

**Expected behavior**
The start should be to left, end to the right, at least for normal ltr text which is the default.

### PR DESCRIPTION
**Pull request by ahocevar**
The Text style's `textAlignment` option takes - among others - `start` and `end` as values. Previously, these values only made sense for right-to-left (RTL) text. With this change, `start` means start of the text for both left-to-right (LTR) and RTL text, and `end` means end.

Replaces #11588
Fixes #11586

### CODE REVIEW

File: src/ol/render/canvas/Executor.js [hunk 1]
```diff
@@ -75,6 +75,20 @@ function getDeclutterBox(replayImageOrLabelArgs) {
   return replayImageOrLabelArgs[3].declutterBox;
 }
 
+const rtlRegEx = /[\u0591-\u07FF]/;
```
Comment (MoonE) [inline src/ol/render/canvas/Executor.js:78]:
```suggestion
const rtlRegEx = new RegExp(
  /* eslint-disable prettier/prettier */
  '[' +
    String.fromCharCode(0x00591) + '-' + String.fromCharCode(0x008ff) +
    String.fromCharCode(0x0fb1d) + '-' + String.fromCharCode(0x0fdff) +
    String.fromCharCode(0x0fe70) + '-' + String.fromCharCode(0x0fefc) +
    String.fromCharCode(0x10800) + '-' + String.fromCharCode(0x10fff) +
    String.fromCharCode(0x1e800) + '-' + String.fromCharCode(0x1efff) +
  ']'
  /* eslint-enable prettier/prettier */
);
```
Apparently it is  not possible to use 5-digit unicode escape codes in js ...

The data I used to look it up is this:
https://github.com/openlayers/openlayers/blob/e7a23d619adc41f8ebe7b57ac5e42fe528533c1c/unicode14.ods?raw=true
sourced from: https://www.unicode.org/Public/14.0.0/ucd/UnicodeData-14.0.0d3.txt

RTL characters are of type `AL` or `R` (orange in the spreadsheet),  LTR characters are of type `L` (blue in the spreadsheet), see: https://www.unicode.org/reports/tr9/#Table_Bidirectional_Character_Types

File: rendering/cases/rtl-text-align/main.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import View from '../../../src/ol/View.js';
+
+const vectorSource = new VectorSource();
+let feature;
+
+// Latin - end (right)
+feature = new Feature({
+  geometry: new Point([-10, 50]),
+});
+feature.setStyle(
```
Comment (MoonE) [inline rendering/cases/rtl-text-align/main.js:20]: Setting a style function on the `VectorSource` and the `text` / `textAlign` on the `Feature`s might make it easier to see what's being tested here.

File: src/ol/render/canvas/Executor.js (hunk 1 above)
Comment (ahocevar) [reply src/ol/render/canvas/Executor.js:78]: Thanks for the thorough research on this!

File: rendering/cases/rtl-text-align/main.js (hunk 2 above)
Comment (ahocevar) [reply rendering/cases/rtl-text-align/main.js]: Good suggestion! Implemented with 1db2c4bc45.

File: rendering/cases/rtl-text-align/main.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      geometry: new Point([10, 0]),
+      text: 'עִברִית',
+      textAlign: 'end',
+    }),
+
+    // Arabic - end (left)
+    new Feature({
+      geometry: new Point([10, -50]),
+      text: 'عربى',
+      font: '24px Ubuntu',
```
Comment (MoonE) [inline rendering/cases/rtl-text-align/main.js:54]: This line can be removed.

**Review by MoonE — APPROVED** [review]
Thanks @ahocevar looks good to me now.

**Pull request merged**